AndroidOnChromeOSDragAndDropDemo/app/src/main/java/dev/hadrosaur/draganddropsample/MainActivity.kt

```diff
@@ -0,0 +1,172 @@
+
+                    if (event.clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)) {
+                        if (item.text != null) {
+                            // Plain text item, show it on screen
+                            binding.textDropTarget.setTextSize(COMPLEX_UNIT_SP, 22f)
+                            binding.textDropTarget.text = item.text
+                        } else {
+                            // This is a plain text file. Read some characters and show them
+                            // Use ContentResolver to resolve the URI
+                            val contentUri = item.uri
+                            var parcelFileDescriptor: ParcelFileDescriptor? = null
+                            try {
+                                parcelFileDescriptor =
+                                    contentResolver.openFileDescriptor(contentUri, "r")
+                            } catch (e: FileNotFoundException) {
+                                e.printStackTrace()
+                                logd("File not found.")
+                                return false
+                            }
+
+                            if (parcelFileDescriptor == null) {
+                                logd("Error: could not load file: ${contentUri.toString()}")
+                                binding.textDropTarget.text = "Error: could not load file: ${contentUri.toString()}"
+                            } else {
+                                // Got the file descriptor, now read the file
+                                val fileDescriptor = parcelFileDescriptor.fileDescriptor
+                                val MAX_LENGTH = 5000
+                                val bytes = ByteArray(MAX_LENGTH)
+
+                                // Read the first MAX_LENGTH bytes of the file
+                                try {
+                                    val `in` = FileInputStream(fileDescriptor)
+                                    try {
+                                        `in`.read(bytes, 0, MAX_LENGTH)
+                                    } finally {
+                                        `in`.close()
+                                    }
+                                } catch (e: Exception) {
+                                    logd("Unable to read file: ${e.message}")
+                                }
+
+                                // Show CHARS_TO_READ chars in the UI
+                                val contents = String(bytes)
+                                val CHARS_TO_READ = 200
+                                val content_length =
+                                    if (contents.length > CHARS_TO_READ) CHARS_TO_READ else 0
+
+                                // Show the read chars in drop target
+                                binding.textDropTarget.setTextSize(COMPLEX_UNIT_SP, 15f)
+                                binding.textDropTarget.text = contents.substring(0, content_length)
```

Comment:
Perhaps it's simpler to do the string processing here as:
```
= contents.substring(0, contents.length.coerceAtMost(MAX_LENGTH))
```

Reply:
Perhaps preface the text with "Dropped text: " and wrap it in quotes? See also line 55 and 111.